src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,135 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of linear maps.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties            as ModuleProperties
+import Algebra.Module.Morphism.Structures   as MorphismStructures
+
+open import Axiom.DoubleNegationElimination
+open import Data.Product
+open import Relation.Binary.Reasoning.MultiSetoid
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+module A = Module modA
+open A using () renaming (Carrierᴹ to A)
+module B = Module modB
+open B using () renaming (Carrierᴹ to B)
+module PA = ModuleProperties modA
+module PB = ModuleProperties modB
+open CommutativeRing ring renaming (Carrier to S)
+
+module _
+  {f : A → B}
+  (isModuleHomomorphism : IsModuleHomomorphism f)
+  where
+
+  open IsModuleHomomorphism isModuleHomomorphism
+
+  -- f(x) ≈ 0 iff x ≈ 0, for linear non-trivial f
+  f0≈0 : f A.0ᴹ B.≈ᴹ B.0ᴹ
+  f0≈0 = begin⟨ B.≈ᴹ-setoid ⟩
+    f A.0ᴹ           ≈⟨ ⟦⟧-cong (A.≈ᴹ-sym (A.*ₗ-zeroˡ A.0ᴹ)) ⟩
+    f (0# A.*ₗ A.0ᴹ) ≈⟨ *ₗ-homo 0# A.0ᴹ ⟩
+    0# B.*ₗ f A.0ᴹ   ≈⟨ B.*ₗ-zeroˡ (f A.0ᴹ) ⟩
+    B.0ᴹ             ∎
+
+  x≈0⇒fx≈0 : ∀ {x} → x A.≈ᴹ A.0ᴹ → f x B.≈ᴹ B.0ᴹ
+  x≈0⇒fx≈0 {x} x≈0 = begin⟨ B.≈ᴹ-setoid ⟩
+    f x    ≈⟨ ⟦⟧-cong x≈0 ⟩
+    f A.0ᴹ ≈⟨ f0≈0 ⟩
+    B.0ᴹ   ∎
+
+  fx≉0⇒x≉0 : ∀ {x} → f x B.≉ᴹ B.0ᴹ → x A.≉ᴹ A.0ᴹ
+  fx≉0⇒x≉0 = contraposition x≈0⇒fx≈0
+
+  -- Zero is a unique output of linear map ≉ `const 0`.
+  zero-unique : ∀ {x} →
+                Σ[ (s , y) ∈ S × A ] ((s A.*ₗ x A.≈ᴹ y) × (f y B.≉ᴹ B.0ᴹ)) →
+                x A.≉ᴹ A.0ᴹ → f x B.≉ᴹ B.0ᴹ
+  zero-unique {x} ((s , y) , (s·x≈y , fy≉0)) x≉0 =
+    PB.s*v≉0⇒v≉0 s·fx≉0
+    where
+    y≉0     : y A.≉ᴹ A.0ᴹ
+    y≉0     = fx≉0⇒x≉0 fy≉0
+    s·fx≉0  : (s B.*ₗ f x) B.≉ᴹ B.0ᴹ
+    s·fx≉0 s·fx≈0 =
+      fy≉0 ( begin⟨ B.≈ᴹ-setoid ⟩
+             f y          ≈⟨ ⟦⟧-cong (A.≈ᴹ-sym s·x≈y) ⟩
+             f (s A.*ₗ x) ≈⟨ *ₗ-homo s x ⟩
+             s B.*ₗ f x   ≈⟨ s·fx≈0 ⟩
+             B.0ᴹ         ∎
+           )
+
+  -- f is odd (i.e. - f (-x) ≈ - (f x)).
```

Comment:
Can't you prove the statement that's in the comment directly? (By linearity, you should be able to pull -1 out. You'd need to first prove that -x is necessarily -1*x, which is useful in its own right).

Reply:
So, the only thing that occurs to me is to show that `(-1 *)` also forms the additive inverse; something like:

```agda
x+[-x]≈x+[-1*x] : ∀ {x} → x A.+ᴹ (A.-ᴹ x) ≈ x A.+ᴹ (- 1#) A.*ₗ x
x+[-x]≈x+[-1*x] = begin⟨ A.≈ᴹ-setoid ⟩
  x A.+ᴹ (A.-ᴹ x) ≈⟨⟩
  A.0ᴹ ≈⟨⟩
  0# A.*ₗ x ≈⟨⟩
  (1# + (- 1#)) A.*ₗ x ≈⟨⟩
  1# A.*ₗ x + (- 1#) A.*ₗ x ≈⟨⟩
  x A.+ᴹ (- 1#) A.*ₗ x
```

But, there are two problems:

1. The code  above is longer than my existing prerequisite: `fx+f[-x]≈0`, and

2. `_+ᴹ_` isn't guaranteed to be injective, or at least I can't find a definition in the library:

    ```agda
    $ git grep -- 'Injective .*_+ᴹ_' src/Algebra/Module/                                 
    
    $
    ```